honeybee_vtk/camera.py

```diff
@@ -0,0 +1,246 @@
+"""A VTK camera object."""
+
+import vtk
+from honeybee_radiance.view import View
+from ladybug_geometry.geometry3d.pointvector import Point3D
+from .actors import Actors
+from ._helper import _check_tuple
+
+
+class Camera(View):
+    """Create a vtk camera object.
+
+        Args:
+            identifier: A text string to be used as a name for the camera.
+                Defaults to 'camera'.
+            position: A tuple of three numbers that represent the x, y and z
+                coordinates of camera in a 3D space. Defaults to (0, 0, 50). Which
+                puts the camera 50 units off of ground (XY plane).
+            direction: A tuple of three numbers that represent the x, y, and z
+                components of a vector towards the aim of the camera.
+                Defaults to (0, 0, -1). Which means the camera will look towards the
+                ground (XY plane).
+            up_vector: A tuple of three numbers to represent the x, y, and z
+                component of the vector that represents where the top of the camera is.
+                Defaults to (0, 1, 0).
+            h_size: A number representing the horizontal view angle.
+                Defaults to 60.
+            v_size: A number representing the vertical view angle.
+                Defaults to 60.
+            type: Choose between a perspective and parallel view type. 'v' will set
+                the perspective view and 'l' will set the parallel view. Defaults to 'v'
+                which is the perspective view.
+        """
+
+    def __init__(self, identifier='camera', position=(0, 0, 50), direction=(0, 0, -1),
+                 up_vector=(0, 1, 0), h_size=60, v_size=60, type='v', bounds=None,
+                 camera_offset=None):
+
+        super().__init__(
+            identifier=identifier, position=position, direction=direction,
+            up_vector=up_vector, h_size=h_size, v_size=v_size, type=type)
+
+        self.bounds = bounds
+        self.camera_offset = camera_offset
+        self._flat_view_directions = {
+                (0, 0, -1): [2, '+'],
+                (0, 0, 1): [2, '-'],
+                (0, 1, 0): [1, '+'],
+                (0, -1, 0): [1, '-'],
+                (1, 0, 0): [0, '+'],
+                (-1, 0, 0): [0, '-'],
+            }
+
+    @property
+    def bounds(self):
+        """Bounds of actors to be used in parallel projection."""
+        return self._bounds
+
+    @bounds.setter
+    def bounds(self, val):
+        if not val:
+            self._bounds = None
+        elif isinstance(val, list) and _check_tuple(val, Point3D):
+            self._bounds = val
+        else:
+            raise ValueError(
+                f'A list of Point3D objects expected. Instead got {val}.'
+            )
+
+    @property
+    def camera_offset(self):
+        """Camera offset is a distance that will be kept between the outermost face of
+        the actors and adjusted camera position in parallel projection."""
```

Comment:
Complied. Camera offset from the properties of a Camera object. Camera offset is the minimum distance that the adjusted camera will maintain from the geometry. For example, say you have a top camera that looks at the ground (XY plane). The highest point in your model is say (5,5,15). The camera offset in this case is the distance that adjusted camera will maintain from this top point. Currently, I am keeping the camera offset distance to 1 unit and I removed the option for a user to change this distance. I think we don't need to expose it but let me know if you think otherwise.